lib/fastlane/actions/update_icloud_container_identifiers.rb

```diff
@@ -0,0 +1,88 @@
+        ubiquity_container_key = 'com.apple.developer.ubiquity-container-identifiers'
+        ubiquity_container_value = result[ubiquity_container_key]
+        raise 'No existing ubiquity container field specified. Please specify an ubiquity container in the entitlements file.'.red unless ubiquity_container_value
+
+        # set iCloud container identifiers
+        result[icloud_container_key] = params[:icloud_container_identifiers]
+        result[ubiquity_container_key] = params[:icloud_container_identifiers]
+
+        # save entitlements file
+        result.save_plist(entitlements_file)
+
+        UI.message "Old iCloud Container Identifiers: #{icloud_container_value}"
+        UI.message "Old Ubiquity Container Identifiers: #{ubiquity_container_value}"
+
+        UI.success "New iCloud Container Identifiers set: #{result[icloud_container_key]}"
+        UI.success "New Ubiquity Container Identifiers set: #{result[ubiquity_container_key]}"
+
+        Actions.lane_context[SharedValues::UPDATE_ICLOUD_CONTAINER_IDENTIFIERS] = result[icloud_container_key]
+      end
+
+      def self.description
+        "This action changes the iCloud container identifiers in the entitlements file"
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :entitlements_file,
+                                       env_name: "FL_UPDATE_ICLOUD_CONTAINER_IDENTIFIERS_ENTITLEMENTS_FILE_PATH", # The name of the environment variable
+                                       description: "The path to the entitlement file which contains the iCloud container identifiers", # a short description of this parameter
+                                       verify_block: proc do |value|
+                                         raise "Please pass a path to an entitlements file. ".red unless value.include? ".entitlements"
+                                         raise "Could not find entitlements file".red if !File.exist?(value) and !Helper.is_test?
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :icloud_container_identifiers,
+                                       env_name: "FL_UPDATE_ICLOUD_CONTAINER_IDENTIFIERS_IDENTIFIERS",
+                                       description: "An Array of unique identifiers for the iCloud containers. Eg. ['iCloud.com.test.testapp']",
+                                       is_string: false,
+                                       verify_block: proc do |value|
+                                         raise "The parameter icloud_container_identifiers need to be an Array.".red unless value.kind_of? Array
+                                       end)
+        ]
+      end
+
+      def self.output
+        ['UPDATE_ICLOUD_CONTAINER_IDENTIFIERS', 'The new iCloud Container Identifiers']
+      end
+
+      def self.authors
+        ["JamesKuang"]
+      end
+
+      def self.is_supported?(platform)
+        platform == :ios
```

Comment:
I think so, but I'm not sure, I don't have a Mac app that uses iCloud to test. This `is_supported` method is the same as from the 'update_app_group_identifiers' action.